doc/en/example/simple.rst

```diff
@@ -895,29 +895,35 @@ here is a little example implemented via a local plugin:
 
     import pytest
 
+    your_unique_key = StashKey[Dict[str, bool]]()
+
 
     @pytest.hookimpl(tryfirst=True, hookwrapper=True)
     def pytest_runtest_makereport(item, call):
         # execute all other hooks to obtain the report object
         outcome = yield
         rep = outcome.get_result()
 
-        # set a report attribute for each phase of a call, which can
+        # store test results for each phase of a call, which can
         # be "setup", "call", "teardown"
-
-        setattr(item, "rep_" + rep.when, rep)
+        if your_unique_key not in item.stash:
+            item.stash[your_unique_key] = {rep.when: rep.passed}
+        else:
+            item.stash[your_unique_key][rep.when] = rep.passed
```

Comment:
Can use `setdefault` to make this more succinct:

```suggestion
        item.stash.setdefault(your_unique_key, {})[rep.when] = rep.passed
```

Reply:
Didn't know this. Thanks.
[ca2cc3d](https://github.com/pytest-dev/pytest/pull/10535/commits/ca2cc3da2a0b218a8ada409c5b4c24baa716ee20)

Also refactored the `tmpdir.py` implementation this way. I can make it to another PR if that's better.
[5959e97](https://github.com/pytest-dev/pytest/pull/10535/commits/5959e979d52e754558af03eb2516a4b43ab5a4e2)

Reply:
> Also refactored the tmpdir.py implementation this way. I can make it to another PR if that's better.

Sure!